benchmarks/timeseries/periodic_feature_extractor.py

```diff
@@ -27,19 +29,23 @@
 from frequenz.sdk.timeseries._quantities import Quantity
 
 
-async def init_feature_extractor(period: int) -> PeriodicFeatureExtractor:
+@contextlib.asynccontextmanager
+async def init_feature_extractor(
+    period: int,
+) -> collections.abc.AsyncIterator[PeriodicFeatureExtractor]:
     """Initialize the PeriodicFeatureExtractor class."""
     # We only need the moving window to initialize the PeriodicFeatureExtractor class.
     lm_chan = Broadcast[Sample[Quantity]]("lm_net_power")
-    moving_window = MovingWindow(
+    async with MovingWindow(
         timedelta(seconds=1), lm_chan.new_receiver(), timedelta(seconds=1)
-    )
-
-    await lm_chan.new_sender().send(Sample(datetime.now(tz=timezone.utc), Quantity(0)))
+    ) as moving_window:
+        await lm_chan.new_sender().send(
+            Sample(datetime.now(tz=timezone.utc), Quantity(0))
+        )
 
-    # Initialize the PeriodicFeatureExtractor class with a period of period seconds.
-    # This works since the sampling period is set to 1 second.
-    return PeriodicFeatureExtractor(moving_window, timedelta(seconds=period))
+        # Initialize the PeriodicFeatureExtractor class with a period of period seconds.
+        # This works since the sampling period is set to 1 second.
+        yield PeriodicFeatureExtractor(moving_window, timedelta(seconds=period))
```

Comment:
looks like this usage of `asynccontextmanager` will mean this function can't be used for creating two simultaneous feature extractor instances, for example,  because when a second call is made,  the first one and its corresponding `MovingWindow` will get cleaned up?

So it has to used only with a `async with` right?

Because I can easily imagine people wanting to create multiple PeriodicFeatureExtractor instances for different data streams, etc. because they might want multiple features.

Reply:
I wonder if it is possible to just make PeriodicFeatureExtractor tie its lifetime with that of its corresponding moving window.  But I guess for that it would need full ownership of the MovingWindow,  which we are not giving.  

People might want to keep using the same MovingWindow instance after or independent of the life time of the PeriodicFeatureExtractor.

Reply:
it was just about what impact the use of `asynccontextmanager` would have if the decorated function has an `async with` inside it, when the decorated function is called multiple times.

Anyway, just wanted to make sure we don't give people the wrong idea about how this can be used,  if people try to read the benchmark script and try to take some inspiration from it.

But it can be examined later and resolved if it is a problem.

Reply:
Oh, ok, if the context manager is used more than once, then it is executed twice and will create a new periodic feature extractor. I think everything should work as expected